iothub/device/src/IotHubBaseClient.cs

```diff
@@ -47,11 +47,18 @@ public abstract class IotHubBaseClient : IDisposable
             if (Logging.IsEnabled)
                 Logging.Enter(this, iotHubClientOptions?.TransportSettings, nameof(IotHubBaseClient) + "_ctor");
 
-            // Make sure client options is initialized.
-            if (iotHubClientOptions == default)
-            {
-                iotHubClientOptions = new();
-            }
+            
+            _clientOptions = iotHubClientOptions != null
```

Comment:
How about making an internal method on the options class to clone it? That way it can be responsible for making a clone and knowing what all fields to copy.

Reply:
Along with some unit tests to ensure all the fields are cloned and not the same instance.